### PR TITLE
Add project categories API documentation

### LINKED ISSUE #418: Document Project Categories API
We should document this endpoint using a combination of:
- [API Blueprint](https://apiblueprint.org/) for writing the documentation of resources, actions, and URIs
- [Dredd](https://dredd.readthedocs.io/en/latest/) for testing the documentation matches the API
- [Apiary](https://apiary.io/) for browsing the docs

Please reference the journey issue #387 for further instructions or commentary on documenting.

### PR DESCRIPTION
**Pull request by begedin**
Closes #418

This one is relatively straightforward, so nothing to say about it.

### CODE REVIEW

**Comment by rigelstpierre** [comment]
Looks good! 🐑 🇮🇹

File: blueprint/api.apib [hunk 1]
```diff
@@ -294,6 +294,48 @@ Write privileges are for admin users only, read can be accessable by any user.
...
++ Response 401 (application/json)
+
+  + Attributes (OAuth Invalid Response)
+
++ Response 422 (application/vnd.api+json)
+
+  + Attributes (Unprocessable Entity Response)
+
+### Delete a Project Category [DELETE /project_categories/:id]
+
```
Comment (joshsmith) [inline blueprint/api.apib:324]: Do you think we should do `## Project Category` above this subheader and use the `:id` URL there? I think that's what we're doing other places.

**Comment by joshsmith** [comment]
👍 other than my one comment.

**Comment by joshsmith** [comment]
Just pushed up changes to my suggestions. Merging now.

**Pull request merged**